resources/assets/js/rich-select.js

```diff
@@ -93,13 +97,15 @@ const RichSelect = (
         this.$refs.button.focus();
     },
     onArrowUp() {
+        const optionsCount = this.getOptionsCount();
         this.selected =
-            this.selected - 1 < 0 ? this.optionsCount - 1 : this.selected - 1;
+            this.selected - 1 < 0 ? optionsCount - 1 : this.selected - 1;
         this.scrollToSelectedOption();
     },
     onArrowDown() {
+        const optionsCount = this.getOptionsCount();
         this.selected =
-            this.selected + 1 > this.optionsCount - 1 ? 1 : this.selected + 1;
+            this.selected + 1 > optionsCount - 1 ? 1 : this.selected + 1;
```

Comment:
this should reset to 0, not 1

```suggestion
            this.selected + 1 > optionsCount - 1 ? 0 : this.selected + 1;
```